aptos-move/aptos-vm/src/move_vm_ext/resolver.rs

```diff
@@ -66,39 +77,18 @@ pub trait MoveResolverExt:
 
     fn get_resource_group(&self, struct_tag: &StructTag) -> Result<Option<StructTag>, VMError> {
         let metadata = self.get_module_metadata(struct_tag.module_id());
-        Ok(Self::get_resource_group_from_metadata(struct_tag, metadata))
-    }
-
-    fn get_resource_group_from_metadata(
-        struct_tag: &StructTag,
-        metadata: Option<aptos_framework::RuntimeModuleMetadataV1>,
-    ) -> Option<StructTag> {
-        metadata.and_then(|metadata| {
-            metadata
-                .struct_attributes
-                .get(struct_tag.name.as_ident_str().as_str())
-                .and_then(|attrs| {
-                    attrs
-                        .iter()
-                        .find_map(|attr| attr.get_resource_group_member())
-                })
-        })
+        Ok(get_resource_group_from_metadata(struct_tag, metadata))
     }
 
     fn is_resource_group(&self, struct_tag: &StructTag) -> bool {
-        let metadata = self.get_module_metadata(struct_tag.module_id());
-        metadata
-            .and_then(|metadata| {
-                metadata
-                    .struct_attributes
-                    .get(struct_tag.name.as_ident_str().as_str())
-                    .and_then(|attrs| {
-                        attrs
-                            .iter()
-                            .map(|attr| Some(attr.is_resource_group()))
```

Comment:
🤦 I hope resource group is the only allowed attributes on structs

Reply:
I don't know precisely what should happen here. I know that the map above can be removed without semantic changes, as we only check if the iter is empty or not.

The only thing I could imagine is that what meant was to find an is_resource_group attribute. Wdyt?

Reply:
I think the fix is correct but cc: @davidiw

Reply:
@davidiw can you comment on this. As in the current version of the code the map on this line has no semantic effect, which smells like a bug.

Reply:
:grimacing: 

Gah... so what we probably want to do is what you did below... have a find statement...

This only works because we only have a single struct attribute at this point in time...

Alternatively we need a find_map :facepalm:

Reply:
yes "find" seemed most natural to me, but wasn't sure.